### PR TITLE
adding tutorial

### PR DESCRIPTION
**Pull request by shenyangHuang**
- [x] how to construct DGraph by raw tensor, pandas, csv
- [x] how to construct DGraph from TGB
- [x] properties of DGraph
- [x] how to dataload 
- [x] how to materialize into batches

### CODE REVIEW

File: docs/tutorials/dgraph.ipynb [hunk 1]
```diff
@@ -0,0 +1,381 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "6a9bfa5c",
+   "metadata": {},
+   "source": [
+    "# Constructing and Accessing Properties in DGraph\n",
```
Comment (Jacob-Chmura) [inline docs/tutorials/dgraph.ipynb]: It's also worth noting (and was my understand to be the preferred user API) that `DGData` design was such the use can directly forward args from the `DGraph` constructor.

For instance, the user can:
```python
# Directly read in csv
dg = DGraph('foo.csv', edge_src_col='foo', edge_dst_col='bar', edge_time_col='time')

# Directly parse a dataframe
dg = DGraph(my_dataframe, ...)
```

Comment (shenyangHuang) [reply docs/tutorials/dgraph.ipynb]: changed the syntax, after merging class constructors to DGraph, this will work now

**Comment by shenyangHuang** [comment]
changed the tutorial to markdown only

File: docs/tutorials/dgraph_tutorial.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
+# Constructing and Accessing Properties in DGraph
+
+This tutorial will show you how to construct a `DGraph` object in `tgm` and accessing its properties.
```
Comment (Jacob-Chmura) [inline docs/tutorials/dgraph_tutorial.md]:
```suggestion
This tutorial shows how to construct a `DGraph` object in `tgm` and explore its properties..
```

File: docs/tutorials/dgraph_tutorial.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+The `DGraph` object can be found in `tgm/graph.py`
```
Comment (Jacob-Chmura) [inline docs/tutorials/dgraph_tutorial.md]:
```suggestion
The `DGraph` class is defined in [`tgm/graph.py`](https://github.com/tgm-team/tgm/blob/main/tgm/graph.py).
```

File: docs/tutorials/dgraph_tutorial.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+
+Let's first learn how to make a `DGraph`by defining the graph ourselves.
```
Comment (Jacob-Chmura) [inline docs/tutorials/dgraph_tutorial.md]: Maybe we should start the tutorial with the simplest and most common use case: move TGB subfolder here, since defining with custom tensors is most advanced construction use case

File: docs/tutorials/dgraph_tutorial.md [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+
+`r` means ordered time granularity which means the underlining timestamps tell us the ordering of edges and we do not use it for time conversion. (other time granularities will allow)
+
+```python
+from tgm.timedelta import TimeDeltaDG
+
+train_dg = DGraph('tgbl-wiki', time_delta=TimeDeltaDG('r'), split='train')
+```
+
+## `DGDataLoader` and `DGHook`
```
Comment (Jacob-Chmura) [inline docs/tutorials/dgraph_tutorial.md]: Should this be part of DGraph tutorial? Seems more advanced than just learning how to construct and query properties on the graph.

**Review by Jacob-Chmura — CHANGES_REQUESTED** [review]
Some minor comments here. There are also some minor aesthetic and formatting that I think could make this look a bit more polished and sharp. 

Do we want to introduce hooks in the intro `DGraph` tutorial? My guess is that hooks are seperate.

TGB should probalbly be explored first since it's the main use case.

Do we want to show users how to discretzie? Or how to slice the graph? These seem more central to the graph view object.

Perhaps device semantics are also worth adding here, since we touch on materialization

File: docs/tutorials/dgraph_tutorial.md [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+import torch
+
+edge_index = torch.LongTensor([[2, 2], [2, 4], [1, 8]])
+edge_timestamps = torch.LongTensor([1, 5, 20])
+edge_feats = torch.rand(3, 5)
+```
+
+## Construct `DGraph` from raw tensors
+
+To construct a `DGraph` object, we use `DGraph.from_raw` method, seen in [`tgm/graph.py`](https://github.com/tgm-team/tgm/blob/main/tgm/graph.py). In TGM, we support both edge events and node events at the same time. The core assumption is that a temporal graph must have edge events and with optional node events as well. You can also easily specify the time granularity of your data with a single string, for example we will specify a second wise granularity, as 's'. Similarly, we can also specify the device.
```
Comment (Jacob-Chmura) [inline docs/tutorials/dgraph_tutorial.md]: We might need to properly introduce the notion of time delta in order for the user to understand how to interpret and use this argument.

Comment (Jacob-Chmura) [reply docs/tutorials/dgraph_tutorial.md]: Using an image from the utg paper could help explain the difference between ctdg and dtdg well

File: docs/tutorials/dgraph_tutorial.md [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+print(f'start time : {our_dgraph.start_time}')
+print(f'end time : {our_dgraph.end_time}')
+print(f'number of nodes : {our_dgraph.num_nodes}')
+print(f'number of edge events : {our_dgraph.num_edges}')
+print(f'number of timestamps : {our_dgraph.num_timestamps}')
+print(f'number of edge and node events : {our_dgraph.num_events}')
+print(f'edge feature dim : {our_dgraph.edge_feats_dim}')
+print(f'static node feature dim : {our_dgraph.static_node_feats_dim}')
+print(f'dynamic node feature dim : {our_dgraph.dynamic_node_feats_dim}')
+print('======================')
```
Comment (Jacob-Chmura) [inline docs/tutorials/dgraph_tutorial.md]: Would be good to clarify how we compute / handle these, especially on subgraphs (e.g. start time is None on empty slice). Number of nodes is based on the max  id + 1.

File: docs/tutorials/dgraph_tutorial.md (hunk 5 above)
Comment (shenyangHuang) [reply docs/tutorials/dgraph_tutorial.md]: for alpha user and complete basic workflow this is the foundation, I would in future PR create separate tutorial for this. Currently this tutorial includes the minimum you need to complete workflow.

File: docs/tutorials/dgraph_tutorial.md (hunk 6 above)
Comment (shenyangHuang) [reply docs/tutorials/dgraph_tutorial.md]: yes, though I think it would be best to be discussed in detail with discretization once we create a separate tutorial for that, I will make issues for to do after alpha

File: docs/tutorials/dgraph_tutorial.md (hunk 4 above)
Comment (shenyangHuang) [reply docs/tutorials/dgraph_tutorial.md]: yes, will discuss `from_tgb` here

File: docs/tutorials/dgraph_tutorial.md (hunk 7 above)
Comment (shenyangHuang) [reply docs/tutorials/dgraph_tutorial.md]: sounds good, added a slicing section too

File: docs/tutorials/dgraph_tutorial.md (hunk 6 above)
Comment (Jacob-Chmura) [reply docs/tutorials/dgraph_tutorial.md]: Sounds good

**Review by Jacob-Chmura — APPROVED** [review]
